docs/reference/combine.md

```diff
@@ -0,0 +1,99 @@
+---
+title: combine
+description:
+nav: 200
+---
+
+# combine
+
+`combine` middleware lets you create a new state by combining an initial state with additional
+state created by a function. This middleware is particularly useful due to infers the state, so you
+don't need to type it.
+
+```js
+combine(initialState, additionalStateCreator)
+```
+
+- [Reference](#reference)
+  - [Signature](#combine-signature)
+  - [`setState` function](#setstate-function)
+  - [`getState` function](#getstate-function)
+  - [`subscribe` function](#subscribe-function)
+  - [`storeApi`](#storeapi)
+- [Usage](#usage)
+- [Troubleshooting](#troubleshooting)
+
+## Reference
+
+### `combine` Signature
+
+```ts
+combine<T, U>(initialState: T, additionalStateCreator: StateCreator<T, [], [], U>): Omit<T, keyof U> & U
```

Comment:
Shouldn't we clarify where we can import this from? `combine` is from `zustand/middleware`.

Reply:
There's a usage section where we can put it